transport/grpc/stream.go

```diff
@@ -133,6 +148,26 @@ func (cs *clientStream) Close(context.Context) error {
 	return cs.stream.CloseSend()
 }
 
+func (cs *clientStream) Headers() (transport.Headers, error) {
+	md, err := cs.stream.Header()
+	if err != nil {
+		return transport.NewHeaders(), err
+	}
+	headers := transport.NewHeadersWithCapacity(len(md))
+	for k, vs := range md {
+		v := vs[0]
+		if len(vs) > 1 {
+			b, err := json.Marshal(vs)
+			if err != nil {
+				return headers, err
+			}
+			v = string(b)
+		}
```

Comment:
This is  accounting for a case which shouldn't happen for yarpc<->yarpc services. Let's just remove this and always choose the first element.

```suggestion
```

Reply:
What should i do in the `>1` case?

Reply:
e.g.

```
		if len(vs) != 1 {
			return headers, yarpcerrors.OutOfRangeErrorf("header key:%q length:%d", k, len(vs))
		}
```

Reply:
I think silently dropping is better. YARPC only supports single header values for a given key, so there's no need to error out if we have some information.